src/osd/OSD.cc

```diff
@@ -7788,6 +7788,21 @@ MPGStats* OSD::collect_pg_stats()
 	min_last_epoch_clean_pgs.push_back(pg->pg_id.pgid);
       });
   }
+
+  {
+    std::lock_guard l(service.pg_delete);
+    auto& osd_deleting_pgs = service.get_osd_deleting_pgs();
+    for (auto& deleting_pg : osd_deleting_pgs) {
+      m->pg_stat[deleting_pg.first] = deleting_pg.second;
+    }
+    auto& osd_deleted_pgs = service.get_osd_deleted_pgs();
+    for (auto& deleted_pg : osd_deleted_pgs) {
+      m->pg_stat[deleted_pg.first] = deleted_pg.second;
+
+      // After sending it to MGR, it could be deleted.
+      osd_deleted_pgs.erase(deleted_pg.first);
```

Comment:
can't modify the container while traversing it!

Reply:
done